docs/how-to/environment/use-environments.md

```diff
@@ -0,0 +1,131 @@
+
+### The default environment 
+
+Hatch will always use the `default` environment if an environment is not chosen explicitly when running a command. 
+
+For instance, the command below builds your project in the default Hatch environment. 
+
+```bash
+> hatch build 
+```
+
+### Customize the default environment 
+
+You can customize the tools that are installed into the Hatch default environment by adding a table called `tool.hatch.envs.default` to your pyproject.toml file. Below is an example of adding the dependencies `pydantic` and `numpy` to the Hatch default environment. 
+
+```toml config-example
+[tool.hatch.envs.default]
+dependencies = [
+  "pydantic",
+  "numpy",
+]
+```
+
+You can declare versions for your dependencies as well within this configuration. 
+
+```toml config-example
+[tool.hatch.envs.default]
+dependencies = [
+  "pydantic>=2.0",
+  "numpy",
+]
+```
+
+### Create custom development environment  
+
+You can create custom environments in Hatch by adding a section to your pyproject.toml file `[tool.hatch.envs.environment-name]`. Below you define an environment called test and you add the pytest dependency to that environment's configuration. 
+
+```toml config-example
+[tool.hatch.envs.test]
+dependencies = [
+  "pytest"
+]
+```
+
+The first time that you call the test environment, hatch will:
+
+1. create the environment
+2. install all dependencies defined in your pyproject.toml file
+3. install your project in editable mode 
+
+### Run commands within a specific environment
+
+Hatch offers a unique environment feature 
+that allows you run a specific command within a specific environment rather than needing to activate the environment as you would using a tool such as conda or venv. 
+
+For instance, if you define an environment called tests that contains the pytest package, you can run
+the pytest command from the tests environment using the syntax:
+
+`hatch -e environment-name:command`
+
+To access the test environment and run pytest, you can run:
+
+```bash
+> hatch run test:pytest
+
+============================== test session starts ===============================
+platform darwin -- Python 3.12.1, pytest-7.4.4, pluggy-1.3.0
+rootdir: /your/path/here/pyosPackage
+collected 0 items   
+```  
+
+Above:
+
+* `test:pytest` represents the name of the environment to call (`test`) and the command to run (`pytest`).
+
+## View current environments
+
+Above you defined and created a new test environment in your pyproject.toml file. You can now 
+use hatch env show to see both the currently created environments and the dependencies in each environment. 
+
+```bach
+> hatch env show
+```
+
+❯ hatch env show
+             Standalone             
+┏━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━┓
+┃ Name    ┃ Type    ┃ Dependencies ┃
+┡━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━┩
+│ default │ virtual │              │
+├─────────┼─────────┼──────────────┤
+│ test    │ virtual │ pytest       │
+└─────────┴─────────┴──────────────┘
```

Comment:
@lwasser Are you running the latest version of Hatch? 
I have another column in this table: "Scripts" 
Did you remove it for simplicity? 
From a diataxis POV, this can trigger anxiety for beginners trying to follow and observe unexplained delta from the how-to.
It's small but worth thinking of. 
Consider this :)

Reply:
I am! 1.9.3. 

so @DahanDv please note that the output of all of this depends on the pyproject.toml file that a user begins with. 

If a user is migrating to hatch they may or may not have that full verbose pyproject.toml that you get with the default `hatch init`. Further, if your config is setup in a way that excludes test matrices when you run hatch init then it also won't have those additional columns / matrix setup.

because in a how-to file, I dont think we can assume a specific user entry point/ I'd rather keep things simple. BUT perhaps it's worth adding a breakout to note your point above about how you may get different results depending upon how you begin to use hatch. In the tutorials we are creating for pyOpenSci we are leading the user through all of the steps and suggesting they won't start with the matrix envt / more complex pyproject.toml just to decrease cognitive load at the beginning. But we will introduce it later. I think for a tutorial exploring all of the ins and outs is a good idea but a how-to should be more concise just giving a user how to do x,y,z without assumptions.

my two cents :)

Reply:
I added:

> [!NOTE]  
> The output may have more columns depending on your environment configuration.